File: go-fiber/app/incoming/taskapi/task_response.go
package taskapi

import (
	"appfiber/outgoing/taskrepo"
)

func NewTaskResponseFromTaskEntity(task taskrepo.TaskEntity) TaskResponse {
	return TaskResponse{
		Description: task.Description,
		Title:       task.Title,
		TaskId:      task.TaskId,
	}
}

File: java-quarkus-reactive/app/src/main/java/science/monke/incoming/TaskResponse.java
package science.monke.incoming;

import com.fasterxml.jackson.annotation.JsonProperty;
import science.monke.outgoing.Task;

import java.util.UUID;

public record TaskResponse(
    @JsonProperty("task_id") UUID taskId,
    @JsonProperty("title") String title,
    @JsonProperty("description") String description) {

  public static TaskResponse fromTask(final Task task) {
    return new TaskResponse(task.taskId, task.title, task.description);
  }
}
